test/metabase/models/interface_test.clj

```diff
@@ -66,3 +72,29 @@
          (with-redefs [mbql.normalize/normalize-tokens (fn [& _] (throw (Exception. "BARF")))]
            ((type-fn :parameters-list :in)
             [{:target [:dimension [:field "ABC" nil]]}]))))))
+
+(deftest timestamped-property-do-not-stomp-on-explicit-values-test
+  (testing "The :timestamped property should not stomp on :created_at/:updated_at if they are explicitly specified"
+    (mt/with-temp Field [field]
+      (testing "Nothing specified: use now() for both"
+        (is (schema= {:created_at java.time.temporal.Temporal
+                      :updated_at java.time.temporal.Temporal
+                      s/Keyword   s/Any}
+                     field))))
+    (let [t        #t "2022-10-13T19:21:00Z"
+          t-schema (s/eq (case (mdb.connection/db-type)
+                           ;; not sure why this is TIMESTAMP WITH TIME ZONE for Postgres but not for H2/MySQL. :shrug:
```

Comment:
This is an H2 bug that's corrected between the version we have and the next one - which is the main thing that makes that migration hard!